app/src/main/java/de/tum/in/tumcampusapp/auxiliary/FavoriteFoodAlarmStorage.java

```diff
@@ -0,0 +1,75 @@
+package de.tum.in.tumcampusapp.auxiliary;
+import android.content.Context;
+import android.content.SharedPreferences;
+import de.tum.in.tumcampusapp.services.FavoriteDishAlarmScheduler;
+
+/**
+ * This class is responsible for keeping track of all the scheduled food alarms.
+ * By instantiating a new entry, it will automatically be added to a hashmap, which
+ * holds all the information about a given day's menus, which are marked as favorite.
+ * If and only if the alarm wasn't already set, an alarm will be scheduled for its date.
+ * This is achieved by instantiating a new instance of FavoriteDishAlarmScheduler, which constructs
+ * a notification, when it's fired.
+ */
+
+public class FavoriteFoodAlarmStorage {
+    private static FavoriteFoodAlarmStorage singleton;
+    private static SharedPreferences sharedPreferences;
+    private static SharedPreferences.Editor editor;
+    private static Context context;
+
+    private FavoriteFoodAlarmStorage(){}
+    public static FavoriteFoodAlarmStorage getInstance(Context context){
+        if (singleton == null){
+            singleton = new FavoriteFoodAlarmStorage();
+        }
```

Comment:
This singleton implementation seems fishy...
Any reason to not use the [InstanceHolder](https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom) Singleton pattern, with synchronized methods?